include/psuedocode.md

```diff
@@ -0,0 +1,54 @@
+The follow outlines some psuedocode for how to implement the logic for a prototype QEO solution.
+
+```
+struct offload_info {
+    uint operation;
+    uint direction;
+    uint next_packet_number;
+    addr ip_address;
+    byte cid[];
+    key_info key;
+}
+
+offload_info[] rx_offload_table;
+offload_info[] tx_offload_table;
+
+int rx_cid_length = -1;
+int tx_cid_length = -1;
+
+void update_offload(offload_info offloads[]) {
+    foreach (auto offload in offloads) {
+        if (offload.operation == "add") {
+            if (offload.direction == "tx") {
+                if (tx_cid_length == -1) tx_cid_length = offload.cid.length;
+                tx_offload_table.add_or_update(offload);
```

Comment:
Oh, is update supposed to be supported? I believe XDP will reject any attempts to update today.

Reply:
Should QEO distinguish between add and update? If not, should we call the operation "set" instead of "add"?

Reply:
Yes, maybe, and sounds good to me.

Reply:
How important is the update scenario for this early prototype? I'd rather not update the XDP QEO implementation until we have settled on requirements to update/delete/identify an offloaded connection.

Reply:
I _think_ we'd be fine limiting this to add only for the prototype. So long as we don't transfer too much and trigger key update scenarios.